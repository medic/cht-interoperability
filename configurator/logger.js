const {format, createLogger, transports} = require('winston');
const isProduction = process.env.NODE_ENV === 'production';

const logger = createLogger({
  level: !isProduction ? 'debug' : 'info',
  format: format.combine(
    format.errors({stack: true}),
    format.splat(),
    format.simple(),
  ),
  transports: [
    new transports.Console({
      // change level if in dev environment versus production
      format: format.combine(
        // https://github.com/winstonjs/winston/issues/1345
        format(info => {
          info.level = info.level.toUpperCase();
          return info;
        })(),
        format.colorize(),
        format.timestamp({
          format: 'YYYY-MM-DD HH:mm:ss',
        }),
        format.printf(
          info =>
          `${info.timestamp} ${info.level}: ${info.message} ${info.stack ? info.stack : ''}`
          )
      ),
    }),
  ],
});

module.exports = logger;
