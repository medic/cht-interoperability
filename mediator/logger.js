const {format, createLogger, transports} = require('winston');
const isProduction = process.env.NODE_ENV === 'production';

const logger = createLogger({
  level: !isProduction ? 'debug' : 'info',
  format: format.combine(
    format.splat(),
    format.simple()
  ),
  transports: [
    new transports.Console({
      format: format.combine(
        format(info => {
          info.level = info.level.toUpperCase();
          return info;
        })(),
        format.colorize(),
        format.timestamp({
          format: 'YYYY-MM-DD HH:mm:ss',
        }),
        format.printf(
          info =>
          `${info.timestamp} ${info.level}: ${info.message} ${info.stack ? info.stack : ''}`
          )
      ),
    }),
  ],
});

module.exports = logger;
