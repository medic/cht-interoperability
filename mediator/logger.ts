const {format, createLogger, transports} = require('winston');
const isProduction = process.env.NODE_ENV === 'production';

export const logger = createLogger({
  level: !isProduction ? 'debug' : 'info',
  format: format.combine(
    format.splat(),
    format.simple()
  ),
  transports: [
    new transports.Console({
      format: format.combine(
        format((info: { level: string; }) => {
          info.level = info.level.toUpperCase();
          return info;
        })(),
        format.colorize(),
        format.timestamp({
          format: 'YYYY-MM-DD HH:mm:ss',
        }),
        format.printf(
          (          info: { timestamp: any; level: any; message: any; stack: any; }) =>
          `${info.timestamp} ${info.level}: ${info.message} ${info.stack ? info.stack : ''}`
          )
      ),
    }),
  ],
});
