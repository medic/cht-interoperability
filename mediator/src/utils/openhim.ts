import { logger } from "../../logger";

export const registerMediatorCallback = (err?: string): void => {
  if (err) {
    throw new Error(`Mediator Registration Failed: Reason ${err}`);
  }

  logger.info('Successfully registered mediator.');
};
