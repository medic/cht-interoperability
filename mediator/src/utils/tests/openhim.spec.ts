import { logger } from '../../../logger';
import { registerMediatorCallback } from '../openhim';

jest.mock('../../../logger');

describe('registerMediatorCallback', () => {
  it('logs to the console when called without an error', () => {
    registerMediatorCallback();

    expect(logger.info).toHaveBeenCalled();
  });

  it('throws an error if an error was passed', () => {
    expect(() =>
      registerMediatorCallback('ERROR')
    ).toThrowErrorMatchingInlineSnapshot(
      `"Mediator Registration Failed: Reason ERROR"`
    );
  });
});
