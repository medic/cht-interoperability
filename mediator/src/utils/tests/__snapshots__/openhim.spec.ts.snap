// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registerMediatorCallback throws an error if an error string is passed in 1`] = `"Mediator Registration Failed: Reason ERROR"`;
